src/state/airdrop/actions.ts

```diff
@@ -11,12 +19,100 @@ export interface ThankYouFlowCompleted {
   type: AirdropAction.ThankYouFlowCompleted;
 }
 
-export type AirdropActionType = ThankYouSeenAction | ThankYouFlowCompleted;
+export interface SubscribeWalletAction {
+  type: AirdropAction.SubscribeWallet;
+  payload: Wallet;
+}
 
-export const markThankYouSeen = (): ThankYouSeenAction => ({
+export interface SubscribeWalletSuccessAction {
+  type: AirdropAction.SubscribeWalletSuccess;
+  payload: {
+    id: string;
+  };
+}
+
+export interface SubscribeWalletFailureAction {
+  type: AirdropAction.SubscribeWalletFailure;
+  error: string;
+}
+
+export interface CheckSubscriptionAction {
+  type: AirdropAction.CheckSubscription;
+  payload: {
+    wallets: Wallet[];
+  };
+}
+
+export interface CheckSubscriptionFailureAction {
+  type: AirdropAction.CheckSubscriptionFailure;
+  error: string;
+}
+
+export interface CheckSubscriptionSuccessAction {
+  type: AirdropAction.CheckSubscriptionSuccess;
+  payload: { subscribedIds: string[] };
+}
+
+export type MarkThankYouSeenActionCreator = () => ThankYouSeenAction;
+
+export const markThankYouSeen: MarkThankYouSeenActionCreator = () => ({
   type: AirdropAction.ThankYouSeen,
 });
 
-export const completeThankYouFlow = (): ThankYouFlowCompleted => ({
+export type CompleteThankYouFlowActionCreator = () => ThankYouFlowCompleted;
+
+export const completeThankYouFlow: CompleteThankYouFlowActionCreator = () => ({
   type: AirdropAction.ThankYouFlowCompleted,
 });
+
+export type SubscribeWalletActionCreator = (payload: Wallet) => SubscribeWalletAction;
+
+export const subscribeWallet: SubscribeWalletActionCreator = payload => ({
+  type: AirdropAction.SubscribeWallet,
+  payload,
+});
+
+export type SubscribeWalletSuccessActionCreator = (id: string) => SubscribeWalletSuccessAction;
+
+export const subscribeWalletSuccess: SubscribeWalletSuccessActionCreator = id => ({
+  type: AirdropAction.SubscribeWalletSuccess,
+  payload: { id },
+});
+
+export type SubscribeWalletFailureActionCreator = (error: string) => SubscribeWalletFailureAction;
+
+export const subscribeWalletFailure = (error: string): SubscribeWalletFailureAction => ({
```

Comment:
so I rename the funciton to creator? `function` doesnt really tell you anything and standard naming for this functions  are `actions creators` so maybe we keep widely used convention instead of creating our own? What cha think?

Reply:
ok can be creator, but please change it everywhere

Reply:
I think you took example from your branch? I dont see any function convention in airdrop or in development.? Hence changed only my implementation.

Reply:
done